torch/distributed/_sharded_tensor/api.py

```diff
@@ -59,6 +78,42 @@ class ShardedTensorMetadata(object):
     memory_format: torch.memory_format
     pin_memory: bool
 
+    def __getstate__(self):
+        # Since torch.memory_format cannot be pickled!
+        if self.memory_format == torch.contiguous_format:
+            mem_format_encoding = 0
+        elif self.memory_format == torch.channels_last:
+            mem_format_encoding = 1
+        elif self.memory_format == torch.preserve_format:
+            mem_format_encoding = 1
```

Comment:
I think I've found a copy-paste error, it seems like from the __setstate__ logic,  mem_format_encoding should be = 2 in this case.